packages/notifications/push/amplify_push_notifications_pinpoint/example/ios/unit_tests/AmplifyPushNotificationsPluginTest.swift

```diff
@@ -0,0 +1,543 @@
+            #selector(getter: MockUIApplicationIconBadgeNumber.applicationIconBadgeNumber)
+        )
+        let originalSetter = class_getInstanceMethod(
+            UIApplication.self,
+            #selector(setter: UIApplication.applicationIconBadgeNumber)
+        )
+        let mockedSetter = class_getInstanceMethod(
+            MockUIApplicationIconBadgeNumber.self,
+            #selector(setter: MockUIApplicationIconBadgeNumber.applicationIconBadgeNumber)
+        )
+        method_exchangeImplementations(originalGetter!, mockedGetter!)
+        method_exchangeImplementations(originalSetter!, mockedSetter!)
+
+        XCTAssertEqual(
+            pluginInstance!.getBadgeCountWithError(fakePointer)?.intValue, fakeIconBadgeNumber
+        )
+
+        let expectedBadgeNumber = 1000
+        pluginInstance!.setBadgeCountWithBadgeCount(expectedBadgeNumber as NSNumber, error: fakePointer)
+
+        XCTAssertEqual(
+            pluginInstance!.getBadgeCountWithError(fakePointer)?.intValue, expectedBadgeNumber
+        )
+
+        method_exchangeImplementations(mockedGetter!, originalGetter!)
+        method_exchangeImplementations(mockedSetter!, originalSetter!)
+    }
+
+    func testApplicationDidBecomeActive() {
+        let original = class_getInstanceMethod(
+            UIApplication.self,
+            #selector(UIApplication.shared.registerForRemoteNotifications as () -> Void)
+        )
+        let mocked = class_getInstanceMethod(
+            MockUIApplicationRegisterForRemoteNotifications.self,
+            #selector(MockUIApplicationRegisterForRemoteNotifications.registerForRemoteNotifications)
+        )
+
+        method_exchangeImplementations(original!, mocked!)
+
+        pluginInstance!.applicationDidBecomeActive(UIApplication.shared)
+
+        XCTAssertTrue(hasCalledRegisterForRemoteNotifications)
+
+        method_exchangeImplementations(mocked!, original!)
+    }
+
+    func testDidReceiveRemoteNotification_AppBackground() {
+        fakeUIApplicationState = UIApplication.State.background
+
+        class MockPushNotificationsFlutterApi: PushNotificationsFlutterApi {
+            override func onNotificationReceivedInBackground(withPayload payload: [AnyHashable : Any], completion: @escaping (Error?) -> Void) {
+                hasOnNotificationReceivedInBackgroundBeenCalled = true
+                XCTAssertTrue(
+                    NSDictionary(dictionary: payload).isEqual(to: expectedNotification)
+                )
+            }
+        }
+
+        let plugin = AmplifyPushNotificationsPlugin(
+            sharedEventsStreamHandlers: EventsStreamHandlers(binaryMessenger: MockFlutterBinaryMessenger()),
+            flutterApi: MockPushNotificationsFlutterApi()
+        )
+        
+
+        let handled = plugin.application(
+            UIApplication.shared,
+            didReceiveRemoteNotification: expectedNotification) { result in
+                XCTAssertEqual(result, .noData)
```

Comment:
With callbacks, you should be using expectations, same as `expectAsync` in Dart.